tools/OutlineService/OutlineService/OutlineService.cs

```diff
@@ -396,6 +376,99 @@ public void ResetRouting(string proxyIp, string proxyInterfaceName)
             RemoveIpv4Redirect();
             RemoveReservedSubnetBypass(proxyInterfaceName);
             StartRoutingIpv6();
+
+            try
+            {
+                StopSmartDnsBlock();
+            }
+            catch (Exception e)
+            {
+                eventLog.WriteEntry($"failed to lift Smart DNS block: {e.Message}",
+                    EventLogEntryType.Warning);
+            }
+        }
+
+        // Disable "Smart Multi-Homed Name Resolution", to ensure the system uses only the
+        // (non-filtered) DNS server(s) associated with the TAP device.
+        //
+        // Notes:
+        //  - To show the current firewall rules:
+        //      netsh wfp show filters
+        //  - This website is an easy way to quickly verify there are no DNS leaks:
+        //      https://ipleak.net/
+        //  - Because .Net provides *no way* to associate the new process with this one, the
+        //    new process will continue to run even if this service is interrupted or crashes.
+        //    Fortunately, since the changes it makes are *not* persistent, the system can, in
+        //    the worst case, be fixed by rebooting.
+        private void StartSmartDnsBlock()

```

Comment:
Is worth trying to stop the service to prevent from running it twice? Relatedly, is it not necessary to create a new `Process` instance?

Reply:
Thinking about this a bit, I decided to use Powershell to kill the process. This means we don't have to re-use the object (and worry about "clearing the streams") and should also be a little more...idempdent if somehow there is >1 instance of the process running.

Reply:
Sounds good.